pkg/cloudprovider/fake/cloudprovider.go

```diff
@@ -262,6 +266,8 @@ func (c *CloudProvider) Name() string {
 	return "fake"
 }
 
-func (c *CloudProvider) GetSupportedNodeClasses() []schema.GroupVersionKind {
-	return c.NodeClassGroupVersionKind
+func (c *CloudProvider) GetSupportedNodeClasses() []status.Object {
+	nodeClasses := make([]status.Object, 1)
+	nodeClasses[0] = status.Object(&v1test1.TestNodeClass{})
+	return nodeClasses
```

Comment:
```suggestion
	return []status.Object{&v1.TestNodeClass{}}
```

I'd just call it v1 or v1alpha1. Best to keep it as close to what an actual version would look like as possible

Reply:
Also, you can simplify all this with a statically declared slice